driver/esp_camera.c

```diff
@@ -171,38 +241,65 @@ static esp_err_t camera_probe(const camera_config_t *config, camera_model_t *out
 #if CONFIG_OV2640_SUPPORT
```

Comment:
We should change these blocks to something like:
```cpp
    case OV2640_PID:
#if CONFIG_OV2640_SUPPORT
         *out_camera_model = CAMERA_OV2640;
         ESP_LOGI(TAG, "Detected OV2640 camera");
         ov2640_init(&s_state->sensor);
#else
         ESP_LOGW(TAG, "Detected OV2640 camera, but support is disabled in menuconfig");
#endif
         break;
```

Reply:
When I saw your message, I changed it to what it is now. Now, if a sensor is not turned on in menuconfig, it will not be detected. 

I think it's OK. Because it is turned on by default, only when the user turns it off will it not be detected, so the user must know that he does not need the sensor.